visualization/app/codeCharta/util/treeMapHelper.ts

```diff
@@ -4,171 +4,190 @@ import { Vector3 } from "three"
-			flattened = s.dynamicSettings.searchedNodePaths.size == 0 ? true : this.isNodeNonSearched(squaredNode, s)
-		}
+function getIncomingEdgePoint(width: number, height: number, length: number, vector: Vector3, mapSize: number) {
+	if (width > length) {
+		return new Vector3(vector.x - mapSize + width / 4, vector.y + height, vector.z - mapSize + length / 2)
+	}
+	return new Vector3(vector.x - mapSize + width / 2, vector.y + height, vector.z - mapSize + length / 4)
+}
 
-		flattened = squaredNode.data.isFlattened || flattened
-		return flattened
+function getOutgoingEdgePoint(width: number, height: number, length: number, vector: Vector3, mapSize: number) {
+	if (width > length) {
+		return new Vector3(vector.x - mapSize + 0.75 * width, vector.y + height, vector.z - mapSize + length / 2)
 	}
+	return new Vector3(vector.x - mapSize + width / 2, vector.y + height, vector.z - mapSize + 0.75 * length)
+}
 
-	private static nodeHasNoVisibleEdges(squaredNode: HierarchyRectangularNode<CodeMapNode>, s: State): boolean {
-		return (
-			squaredNode.data.edgeAttributes[s.dynamicSettings.edgeMetric] === undefined ||
-			s.fileSettings.edges.filter(edge => squaredNode.data.path === edge.fromNodeName || squaredNode.data.path === edge.toNodeName)
-				.length == 0
-		)
+function isNodeToBeFlat(codeMapNode: CodeMapNode, s: State): boolean {
+	let flattened = false
+	if (s.appSettings.showOnlyBuildingsWithEdges && s.fileSettings.edges?.filter(edge => edge.visible).length > 0) {
+		flattened = nodeHasNoVisibleEdges(codeMapNode, s)
 	}
 
-	private static isNodeNonSearched(squaredNode: HierarchyRectangularNode<CodeMapNode>, s: State): boolean {
-		return !s.dynamicSettings.searchedNodePaths.has(squaredNode.data.path)
+	if (s.dynamicSettings.searchedNodePaths && s.dynamicSettings.searchPattern && s.dynamicSettings.searchPattern.length > 0) {
+		flattened = s.dynamicSettings.searchedNodePaths.size == 0 ? true : isNodeNonSearched(codeMapNode, s)
 	}
 
-	private static getBuildingColor(node: CodeMapNode, s: State, isDeltaState: boolean, flattened: boolean): string {
-		const mapColorPositive = s.appSettings.whiteColorBuildings ? s.appSettings.mapColors.lightGrey : s.appSettings.mapColors.positive
-		if (isDeltaState) {
-			return s.appSettings.mapColors.base
-		} else {
-			const metricValue: number = node.attributes[s.dynamicSettings.colorMetric]
-
-			if (metricValue == null) {
-				return s.appSettings.mapColors.base
-			} else if (flattened) {
-				return s.appSettings.mapColors.flat
-			} else if (metricValue < s.dynamicSettings.colorRange.from) {
-				return s.appSettings.invertColorRange ? s.appSettings.mapColors.negative : mapColorPositive
-			} else if (metricValue > s.dynamicSettings.colorRange.to) {
-				return s.appSettings.invertColorRange ? mapColorPositive : s.appSettings.mapColors.negative
-			} else {
-				return s.appSettings.mapColors.neutral
-			}
-		}
+	flattened = codeMapNode.isFlattened || flattened
```

Comment:
Please move the `codeMapNode.isFlattened` condition to the top and return right away if it's truthy. Otherwise all other code is redundant.

```js
if (codeMapNode.isFlattened) {
  return true
}
...
```